### PR TITLE
Add e2e tests for Helm support (#93)

### PR DESCRIPTION
**Pull request by nan-yu**
Add more e2e tests to verify `gcenode` and `gcpserviceaccount` auth types work with Artifact Registry.

### CODE REVIEW

**Comment by nan-yu** [comment]
/hold

Need to rebase after #143 is merged.

**Comment by tiffanny29631** [comment]
QQ: are we going to have a v1.13.1 branch or we just keep the latest patch in v1.13?

**Comment by nan-yu** [comment]
one single `v1.13` branch for all patch releases.

**Review by xinnywinne — APPROVED** [review]
/lgtm

**Pull request merged**
